middle_end/flambda2/to_cmm/to_cmm_primitive.ml

```diff
@@ -151,6 +151,38 @@ let array_load ~dbg (kind : P.Array_kind.t) ~arr ~index =
   | Naked_floats -> C.unboxed_float_array_ref arr index dbg
   | Naked_int32s -> C.unboxed_int32_array_ref arr index dbg
 
+let array_vector_load ~dbg (vec_kind : Lambda.boxed_vector)
+    (array_kind : P.Array_kind.t) ~arr ~index =
+  P.Array_kind.check_vector_access_kind array_kind vec_kind;
+  let offset =
```

Comment:
maybe rename that to a more explicit name, such as `element_width_log2` ? Also, it took a bit of time for @lthls and I to understand the semantics of these array operations, it'd be nice if they were documented somewhere (for instance in the primitive file, or in a separate doc somewhere)

Reply:
Did this question about documentation get resolved?

Reply:
I don't think so. It'd probably deserves a file in the `docs` folder.

Reply:
I've added `middle_end/flambda2/docs/simd.md`, that sufficient for now?

Reply:
that should be good for now